libraries/lib-module-manager/AsyncPluginValidator.h

```diff
@@ -0,0 +1,64 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file AsyncPluginValidator.h
+
+  @author Vitaly Sverchinsky
+
+  Part of lib-module-manager library
+
+**********************************************************************/
+
+#pragma once
+
+#include <wx/string.h>
+#include <memory>
+
+class PluginDescriptor;
+
+
+/**
+ * \brief Used to perform plugin validation in a dedicated process.
+ * Once instantiated, client should call AsyncPluginValidation::Validate
+ * for plugin that needs to be validated. When done,
+ * AsyncPluginValidation will notify caller via Delegate.
+ */
+class MODULE_MANAGER_API AsyncPluginValidator final
+{
+   class Impl;
+   std::shared_ptr<Impl> mImpl;
+public:
+   /**
+    * \brief Used to talk back to calling side
+    */
+   class MODULE_MANAGER_API Delegate
+   {
+   public:
+      virtual ~Delegate();
+
+      ///Called for each plugin instance found inside module
+      virtual void OnPluginFound(const PluginDescriptor& plugin) = 0;
+      ///Called when module processing finished
```

Comment:
This means, handling of one request is complete, and another may be sent.

It is very pleasing to see only one virtual function interface here, and the simplified implementation without the state machine and a switch.  I find it more understandable than the first draft.